src/config.h

```diff
@@ -307,4 +307,10 @@ void setcpuaffinity(const char *cpulist);
 #define HAVE_FADVISE
 #endif
 
+#if (defined(__GNUC__) && __GNUC__ >= 7) || (defined(__clang__) && __clang_major__ >= 7)
+#if defined(__x86_64__)
```

Comment:
i don't think we need this one, i think we can rely on `__builtin_cpu_supports` (which i assume will be optimized at compile time in case the compiler is sure about the outcome).
maybe someone can validate that?